### PR TITLE
Feat: issue#63 - GET /user/personal_details api endpoint and tests

### LINKED ISSUE #62: Development: Create/Update a profile on BIT
**Is your feature request related to a problem? Please describe.**
As a student, I need to have the ability to create a profile on BIT. For this the API on the backend should be available.

**Describe the solution you'd like**
Create an API for profile creation/updation for the BIT backend. The API would take profile detail combinations as input and return a response code based on the validity of the data in the profile information. Majorly, 2 outcomes are possible: Successful (200 OK), or Unsuccessful (40x) (x represents one of the error codes like 400, 403, etc.), Not found (404), 500 (internal server error)

### LINKED ISSUE #72: Dev: GET /user/personal_details api endpoint and tests
## Description
As a student,
I need to create GET /user/personal_details endpoint and tests,
so that I can give user the ability to get their Personal Details.

## Mocks
as per MVP mockup

Screen Shot 2020-07-09 at 1 23 52 am

### Update [Required]
- [ ] User can update their Personal details profile
- [ ] have these test cases: 
    * 200 - Success
    * 40x - can be Bad Request, Not Found or Unauthorized
    * 500 - Internal Server Error

### Enhancement to Update [Optional]
NA

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 day

**Additional Note**
This issue is part of issue #63 GET /user profile

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description

Add API endpoint for GET/user profile and test cases

Fixes #72 as well as part of issue #63 

### Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- [x] tested functionality by running the application manually
- [x] run test on VSCode Tests explorer - result: all passed
- [x] run unittest on the terminal - result: all passed

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/api/ms_api_utils.py [hunk 1]
```diff
@@ -1,25 +1,17 @@
...
-# WARNING!!! When you push a PR, for travis to pass test cases related to MS API
-# Heroku MS API needs to be set as preference over the localhost. Otherwise, make sure
-# you run the MS API local server when you push the PR.
-# BASE_MS_API_URL = "https://bridge-in-tech-ms-test.herokuapp.com"
-
-# create instance
-
-def post_request(request_url, data):
+def post_request(request_string, data):
+    request_url = f"{BASE_MS_API_URL}" + request_string
```
Comment (ramitsawhney27) [inline app/api/ms_api_utils.py]: Simplify as:
f”{BASE_MS_API_URL}{request_string}”

**Review by ramitsawhney27 — DISMISSED** [review]

File: app/api/ms_api_utils.py (hunk 1 above)
Comment (mtreacy002) [reply app/api/ms_api_utils.py]: done.

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've fixed the title of this PR. It should be addressing issue #63  instead of #62. I'll do #62 next.

**Comment by mtreacy002** [comment]
PS: I need to open a new branch for issue #62 and will need to carry this PR code changes across. I'll squashed merge it now so it won't be a problem later on.

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've just forced push last fix for f-string referred by @ramitsawhney27 [above](https://github.com/anitab-org/bridge-in-tech-backend/pull/71#discussion_r451015626) that I missed (on get_request function).

**Comment by mtreacy002** [comment]
@ramitsawhney27 . Atm, this code change on this PR only getting user from user table on MS schema. do you want me to also add the do the BIT schema DAO side on this PR as well or open a new one for that?

Screen Shot 2020-07-08 at 11 32 59 pm

**Comment by mtreacy002** [comment]
But perhaps, the BIT DAO can be done after update user profile, since BIT DAO will only have data after user update their profile which is for issue #62. I'll leave this PR as it is then (only GET /user on MS public) and will open a separate issue for GET /user that calls BIT DAO user_extension and personal_background

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers. As per MVP mockups on User `My Profile` page, I'll separate GET, PUT, DELETE for user profile into 3 sub-urls: 
1. `/user/personal_details` (data source: User table on public (MS) schema)
Screen Shot 2020-07-09 at 1 23 52 am

2. `/user/additional_information` (data source: User_extension table on bitschema)
Screen Shot 2020-07-09 at 1 24 03 am

3. `/user/personal_background` (data_source: Personal_background table on bitschema)
Screen Shot 2020-07-09 at 1 24 19 am

This PR will only deal with GET /user/personal_details. I'll create separate issues for the other 2 GET sub-urls.

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I've just pushed forced the changes to fix request url `/user` to `/user/personal_details` as per discussion above.

**Comment by mtreacy002** [comment]
Now this PR is complete and any changes will be made in the next PR. Please re-review. @anitab-org/bridgeintech-maintainers

**Comment by mtreacy002** [comment]
Update: just came across a bug on this PR... will fix and push another commit soon @anitab-org/bridgeintech-maintainers

**Comment by mtreacy002** [comment]
Update @anitab-org/bridgeintech-maintainers . I fixed the bug. Now it's ready for re-review 😉

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done by @ramitsawhney27 

2. All possible responses were tested as below (tap/zoom in to see clearer gifs):

| **GET /user/personal_details on BIT API**  | **Expected Result** | **Actual Result** |
|---|---|---|
|   test-pr-bit-backend-71-success | As a user, I am able to view my personal details when I enter a correct access token.   |   ✅   |
|  test-pr-bit-backend-71-invalid token |   As a user, I will get a `token invalid error` when I enter an incorrect access token.   |   ✅   |
| test-pr-bit-backend-71-internal-server-error |   As a user, I will get an `internal server error` when I stop running MS backend locally |   ✅   |

3. Additional testcases covered: N/A
    
4. Additional Comments: 
* Tested locally with no errors @mtreacy002 !! 

5. OS Version: Windows 10

**Comment by mtreacy002** [comment]
Thanks for the review, @foongminwong. @ramitsawhney27 . can you please re-approve this PR? your previous approval got dismissed when I did the force pushed 😁

**Review by ramitsawhney27 — APPROVED** [review]

**Pull request merged**